bluesky/traffic/route.py

```diff
@@ -898,20 +849,19 @@ def newcalcfp(self):
 
         # Insert actual position as A/C waypoint
         idx = self.iactwp
-        self.insertcalcwp(self,idx,"A/C")
+        self.insertcalcwp(idx,"A/C")
         self.wplat[idx] = bs.traf.lat[self.iac] # deg
         self.wplon[idx] = bs.traf.lon[self.iac] # deg
         self.wpalt[idx] = bs.traf.alt[self.iac] # m
         self.wpspd[idx] = bs.traf.tas[self.iac] # m/s
 
         # Calculate distance to last waypoint in route
-        dist2go = []
-        nwp = len(wpname)
+        nwp = len(self.wpname)
         dist2go = [0.0]
-        for i in range(nwp-2,-1,-1):
-            qdr,dist = geo.qdrdist(self.wplat[i],self.wplon[i],    \
-                        self.wplat[i+1],self.wplon[i+1])
-            dist2go = [dist2go[i+1]+dist]+dist2go
+        for i in range(nwp - 2, -1, -1):
+            qdr, dist = geo.qdrdist(self.wplat[i], self.wplon[i],
+                                    self.wplat[i + 1], self.wplon[i + 1])
+            dist2go = [dist2go[0] + dist] + dist2go

```

Comment:
Bug: original code (index i+1) breaks. Should be 0 - last distance is always at start of list when building up. Index i+1 does not exist - IndexError.